### PR TITLE
Rename reinitialize -> reset and use deny(unused_variables) pattern

### PR DESCRIPTION
**Pull request by AaronM04**

### CODE REVIEW

**Comment by Manghi** [comment]
LGTM! Nice application of that blog post 😄 
Will need to be updated due to changes in #6. I propose we merge this in after #6 is merged.

Edit: Looks like I hadn't pushed changes related to clearing out the queues. Look at  6763063  to see what I mean.

**Review by Manghi — APPROVED** [review]

**Comment by AaronM04** [comment]
Thanks! :smile:

**Pull request merged**
